src/codecs/webp/extended.rs

```diff
@@ -0,0 +1,683 @@
+use std::convert::TryInto;
+use std::io::{self, Error, Read};
+use std::{error, fmt};
+
+use super::decoder::{read_chunk, WebPRiffChunk};
+use super::lossless::{LosslessDecoder, LosslessFrame};
+use super::vp8::{Frame as VP8Frame, Vp8Decoder};
+use crate::error::DecodingError;
+use crate::image::ImageFormat;
+use crate::{color, Delay, Frame, Frames, ImageError, ImageResult, Rgba, RgbaImage};
+use byteorder::{LittleEndian, ReadBytesExt};
+
+//all errors that can occur while parsing extended chunks in a WebP file
+#[derive(Debug, Clone, Copy)]
+enum DecoderError {
+    ChunkInvalid,
```

Comment:
Can you add the `WebPRiffChunk` identification as data here, and include it in the error message? This is simple yet possibly effective at identifying a little more information. I wouldn't mind using the chunk's `fmt::Debug` information in the error's `Display` actually.